Source/WebInspectorUI/UserInterface/Views/FontDetailsPanel.js

```diff
@@ -521,6 +536,8 @@ WI.FontDetailsPanel = class FontDetailsPanel extends WI.StyleDetailsPanel
 
     _handleFontVariationValueChanged(event)
     {
+        this._skipNextUpdate = true;
+        this._debouncedUpdate.delayForTime(250);
```

Comment:
A quarter of a second is an awfully long time between updates IMO when trying to view changes on the page. Long enough I'd think that I'd overshoot where I want to land with the slider if I'm trying to drag to a specific value. Is performance acceptable around 50ms or 100ms?

Reply:
Updates to the page styles always happen live. They are not debounced or throttled.
What we're handing in this patch is the Fonts panel **reacting to style changes**, some of which it originates.

> Long enough I'd think that I'd overshoot where I want to land with the slider if I'm trying to drag to a specific value.

The values in the Fonts panel, Styles panel and what's visible on page remain in sync throughout slider interaction. There's no chance of overshooting.

The Fonts panel value in the DOM is updated locally by the slider's `input` event handler before dispatching the event to apply the styles to the page. 

`FontVariationDetailsSectionRow.js:61-64`
```js
this._inputRangeElement.addEventListener("input", (event) => {
    this.value = event.target.value;
    this.dispatchEventToListeners(WI.FontVariationDetailsSectionRow.Event.VariationValueChanged, {tag: event.target.name, value: event.target.value});
}, {signal: abortSignal});
 ```
 
Here's what's going on while dragging a slider:

```
this._skipNextUpdate = true
```
Flipping this flag inhibits updating the Fonts panel DOM and doing associated `FontStyles` computations *in reaction to changes* we're fairly certain we just caused.

```
this._debouncedUpdate.delayForTime(250)
```
A short time after the slider stops changing (`250ms` in this case; could be lowered) we force an update just in case there's a style change that occurred in the page after the Fonts panel stopped making changes. This is the "failsafe" we discussed to ensure we catch any updates by in-page scripting.

In the debounced operation, the `this._skipNextUpdate` flag is flipped back off. This allows the Fonts panel to update in reaction to changes coming from the outside. These updates are throttled.